### PR TITLE
Demonstrate issue 3233

### PR DESCRIPTION
**Pull request by sungam3r**
#3233 

Just to explicitly state the current status quo with serialization.

### CODE REVIEW

**Review by Shane32 — APPROVED** [review]

**Comment by Shane32** [comment]
What about changing ErrorInfoProvider so that data is not included by default?

**Comment by Shane32** [comment]
> Do you mean `ErrorInfoProviderOptions.ExposeData` ? `ErrorInfoProvider` just uses its boolean value. I would not change ` = true` default for `ExposeData`.

Yes.  I really don't know why anyone would ever want `ExposeData` to be `true`, any more than they want to reveal stack traces.

**Comment by sungam3r** [comment]
😕 I think we use `true` as default value because a long time ago it was used in old 2.4.0 version.

**Comment by sungam3r** [comment]
In our backend we use this data dictionary inside error to control some specific auth behaviour across microservices.

**Comment by sungam3r** [comment]
I want to say that this data has nothing to do with stack traces. It is some sort or user-defined structured information.

**Comment by Shane32** [comment]
> I want to say that this data has nothing to do with stack traces. It is some sort or user-defined structured information.

But I believe the most common use case is internal information within C#. For example the MS SQL library adds data here. Somehow Azure is adding data there causing the problem related to this PR. In both of those cases, it is often not the intent of the developer to expose such data to the end user. 

I am not arguing that certain users, such as yourself, may wish to use the data field. But it seems the exception and not the rule.

To summarize, it seems the data field is used by other services and often unintentionally exposed to a caller of the API.

**Comment by sungam3r** [comment]
I am inclined to agree. Do you want to change default?

**Comment by Shane32** [comment]
> I am inclined to agree. Do you want to change default?

Completed in #3263
